metabase/metabase
# Convert TagEditorParam-related logic to TypeScript and add tests

Epic https://github.com/metabase/metabase/issues/29881

Changes:
- Adds fingerprints to `sample_database` mock preset
- Moves `TemplateTag` types to `metabase-types/api`
- Converts `tag-template-options` to TS 
- Converts tests for `TagEditorParam` to TS

How to verify:
- `yarn type-check` and `yarn test-unit` passes


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29411)
